### PR TITLE
Wait for criuProcess once

### PR DESCRIPTION
**Pull request by tedyu**
In linuxContainer#criuSwrk(), the deferred func may wait on criuProcess for a second time, if the following has been executed (around line 1640):
```
	st, err = criuProcess.Wait()
```
This PR adds check for st so that there is no redundant wait.

Note: there is already deferred criuClientCon.Close() on line 1476

### CODE REVIEW

**Comment by kolyshkin** [comment]
1. I do not understand the meaning of
```go
                if err != nil {
                        return
                }
```
in the `defer func()`. Looks like it does nothing, so it does not make sense to have it at all (unless I am mistaken and it actually means something).

2. Since we are ignoring the error, there is nothing wrong to call Wait for a second time.

3. I would remove the second "criuClientCon.Close()" (as a separate patch, first in the series).

**Comment by kolyshkin** [comment]
@avagin PTAL

**Comment by tedyu** [comment]
w.r.t. #1, I have updated the PR to log the error.

File: libcontainer/container_linux.go [hunk 1]
```diff
@@ -1504,11 +1504,14 @@ func (c *linuxContainer) criuSwrk(process *Process, req *criurpc.CriuReq, opts *
...
 	defer func() {
-		criuClientCon.Close()
-		_, err := criuProcess.Wait()
-		if err != nil {
-			return
+		if st == nil {
+			criuClientCon.Close()
+			_, err := criuProcess.Wait()
+			if err != nil {
+				logrus.Warningf("wait on criuProcess returned %v", err)
```
Comment (kolyshkin) [inline libcontainer/container_linux.go:1513]: nit: `s/Warningf/Warnf/` (both are OK but looks like `Warnf` is used everywhere else in runc code).

Comment (tedyu) [reply libcontainer/container_linux.go]: Addressed.

**Comment by tedyu** [comment]
Ping @avagin

File: libcontainer/container_linux.go [hunk 2]
```diff
@@ -1504,11 +1504,14 @@ func (c *linuxContainer) criuSwrk(process *Process, req *criurpc.CriuReq, opts *
 	// cmd.Process will be replaced by a restored init.
 	criuProcess := cmd.Process
 
+	var st *os.ProcessState
```
Comment (avagin) [inline libcontainer/container_linux.go:1507]: nit: this name is good when is used in a block of 20-lines code, but in this case, I would use something more descriptive (e g criuProcessState).

**Comment by avagin** [comment]
LGTM

File: libcontainer/container_linux.go (hunk 2 above)
Comment (tedyu) [reply libcontainer/container_linux.go]: Your comment has been addressed.

**Comment by kolyshkin** [comment]
LGTM

**Comment by kolyshkin** [comment]
@AkihiroSuda @mrunalp PTAL

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**